corehq/form_processor/submission_post.py

```diff
@@ -112,6 +113,8 @@ def _set_submission_properties(self, xform):
         if self.date_header:
             xform.date_header = self.date_header
 
+        if self.build_profile_id:
```

Comment:
I don't think this is necessary - you could just always set it

Reply:
thought of following the same pattern as above for other properties. Agree that it is not necessary.

Reply:
https://github.com/dimagi/commcare-hq/pull/25816/commits/8d41e8837559cfcb818f913c80a9845c647b7dcd